Document limitations for Select ... lock() (#630)
cap-js/docs: java/query-api.md

Select.from("bookshop.Books").byId(1).lock(SHARED);
```

> Not every database entity exposed via the CDS entity can be locked with the `lock()` clause. Databases require that the target of such statements is represented by a single table or a view that is based on a single table without aggregation, joining or coalescence of values and the statement itself must not introduce aggregations, for example, by adding `groupBy()` clauses. Some databases might have additional restrictions or limitations specific to them.  

Reviewer: ```suggestion
Not every entity exposed via the CDS entity can be locked with the `lock()` clause. Databases require that the target of such statements is represented by a single table or by a view that is simple enough so that the database can ambiguously identify which rows must be locked. On views, which are joining multiple tables, which are aggregating data, or which are computing expressions or coalescence of values `lock()` can't be executed. Some databases might have additional restrictions or limitations specific to them.  
```

There are few notable examples of such restrictions: 

> * You cannot use `lock()` and `distinct()` in the same statement.

Reviewer: ```suggestion
* You cannot use `lock()` with `distinct()` or `groupBy()`.
```

> * Localized entities can be locked only if you manually reset the locale via dedicated request context without the locale as described in the chapter [Modifying Request Context](./request-contexts#modifying-requestcontext). 

Reviewer: ```suggestion
* Localized entities can be locked only if you the query is executed without a locale as described in the chapter [Modifying Request Context](./request-contexts#modifying-requestcontext). 
```
